### PR TITLE
Adds progressbar for glb, changes names and layout of scale mean and sd, improves error handling

### PR DESCRIPTION
**Pull request by juliuspf**
fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1236
fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1379

the other changes are inline with what @AlexanderLyNL requested

### CODE REVIEW

**Comment by juliuspf** [comment]
Layout would be this: 
Screenshot 2021-05-14 at 12 08 03

**Comment by juliuspf** [comment]
I heard back from Klaas Sijtsma, he thinks the terminology is fine, he thought the following layout was clearer ("more concrete") though: 
Screenshot 2021-05-17 at 10 11 21

I do not agree with that, I thought the previous option was clear, but I have no idea what users will expect really. What do you think @AlexanderLyNL @vandenman

**Comment by AlexanderLyNL** [comment]
I like the fact that this layout 

> Screenshot 2021-05-14 at 12 08 03

looks more clean to me. I guess that the major difference is that you typically first think about whether you're interested in sum scores or mean scores and then consider the means and sds. Either layout will work for me, since you cannot show the mean of sum scores and the sd of mean scores.

**Comment by vandenman** [comment]
I don't care too much, but I like the original layout slightly more.

**Comment by juliuspf** [comment]
Given that the original layout is the one implemented in this PR so far, I think we can leave it at that :) So I suppose you can review @vandenman

File: R/uniDimBayesianCalcCoeff.R [hunk 1]
```diff
@@ -306,17 +307,24 @@
...
     if (is.null(out[["samp"]])) {
-      startProgressbar(model[["progressbarLength"]] %/% 500 + 1)
-      out[["samp"]] <- coda::mcmc(apply(model[["gibbsSamp"]], MARGIN = c(1, 2), Bayesrel:::glbOnArray_custom,
-                                        callback = progressbarTick))
+
+      dd <- dim(model[["gibbsSamp"]])
+      out[["samp"]] <- matrix(0, dd[1], dd[2])
+
+      startProgressbar(dd[1] * 3)
+      for (i in 1:dd[1]) {
```
Comment (vandenman) [inline R/uniDimBayesianCalcCoeff.R:315]:
```suggestion
      for (i in seq_len(dd[1])) {
```
I also wouldn't mind if you use `nrow` and `ncol` directly, although that is slightly more verbose. Up to you.

File: R/uniDimBayesianCalcCoeff.R [hunk 2]
```diff
@@ -508,20 +517,20 @@
       return(out)
 
     stateContainer <- .getStateContainerB(jaspResults)
-    stateContainer[["itemMeanObj"]] <- createJaspState(out, dependencies = c("itemMean"))
+    stateContainer[["meanItemObj"]] <- createJaspState(out, dependencies = c("meanItem"))
```
Comment (vandenman) [inline R/uniDimBayesianCalcCoeff.R:520]:
```suggestion
    stateContainer[["meanItemObj"]] <- createJaspState(out, dependencies = "meanItem")
```

File: R/uniDimBayesianCalcCoeff.R [hunk 3]
```diff
@@ -530,7 +539,7 @@
       return(out)
 
     stateContainer <- .getStateContainerB(jaspResults)
-    stateContainer[["itemSdObj"]] <- createJaspState(out, dependencies = c("itemSd"))
+    stateContainer[["sdItemObj"]] <- createJaspState(out, dependencies = c("sdItem"))
```
Comment (vandenman) [inline R/uniDimBayesianCalcCoeff.R:542]:
```suggestion
    stateContainer[["sdItemObj"]] <- createJaspState(out, dependencies = "sdItem")
```

File: R/uniDimFrequentistCalcCoeff.R [hunk 4]
```diff
@@ -431,7 +431,7 @@
       for (i in 1:model[["k"]]) {
```
Comment (vandenman) [inline R/uniDimFrequentistCalcCoeff.R:431]:
```suggestion
      for (i in seq_len(model[["k"]])) {
```

File: R/uniDimFrequentistCalcCoeff.R [hunk 5]
```diff
@@ -569,43 +569,43 @@
...
-  if (options[["itemMean"]]  && is.null(model[["empty"]])) {
+  if (options[["meanItem"]]  && is.null(model[["empty"]])) {
     out[["itemDropped"]] <- colMeans(dataset, na.rm = TRUE)
 
     if (options[["disableSampleSave"]])
       return(out)
 
     stateContainer <- .getStateContainerF(jaspResults)
-    stateContainer[["itemMeanObj"]] <- createJaspState(out, dependencies = c("itemMean"))
+    stateContainer[["meanItemObj"]] <- createJaspState(out, dependencies = c("meanItem"))
```
Comment (vandenman) [inline R/uniDimFrequentistCalcCoeff.R:587]:
```suggestion
    stateContainer[["meanItemObj"]] <- createJaspState(out, dependencies = "meanItem")
```

File: R/uniDimFrequentistCalcCoeff.R [hunk 6]
```diff
@@ -569,43 +569,43 @@
...
+  if (options[["sdItem"]]  && is.null(model[["empty"]])) {
 
     out[["itemDropped"]] <- apply(dataset, 2, sd, na.rm = TRUE)
 
     if (options[["disableSampleSave"]])
       return(out)
 
     stateContainer <- .getStateContainerF(jaspResults)
-    stateContainer[["itemSdObj"]] <- createJaspState(out, dependencies = c("itemSd"))
+    stateContainer[["sdItemObj"]] <- createJaspState(out, dependencies = c("sdItem"))
```
Comment (vandenman) [inline R/uniDimFrequentistCalcCoeff.R:608]:
```suggestion
    stateContainer[["sdItemObj"]] <- createJaspState(out, dependencies = "sdItem")
```

**Review by vandenman — CHANGES_REQUESTED** [review]
The changes look fine. I left some mainly stylistic comments below. If you ever want to find these yourself, you could try [lintr](https://github.com/jimhester/lintr). That's a static code analysis tool that can check for instance, if you use `seq_len` instead of `1:...` (seq_linter) and it can yell at you if you do something like `c()` with only one item (unneeded_concatenation_linter). See https://github.com/jimhester/lintr#available-linters for an overview. You don't need to follow all possible linters btw, just follow common sense.

Even though the changes are minimal and the code looks fine, I'd prefer to wait until the unit tests are functional again before merging this.

File: R/uniDimFrequentistCalcCoeff.R (hunk 6 above)
Comment (juliuspf) [reply R/uniDimFrequentistCalcCoeff.R]: thanks for the tip. I implemented the changes, however, the liner package did not complain about the unneeded concatenation somehow. I removed it anyhow.

Comment (vandenman) [reply R/uniDimFrequentistCalcCoeff.R]: it's might not be a default lintr, it could be that you need to specify that check manually.

**Comment by vandenman** [comment]
should not be merged until @juliuspf is ready

**Comment by juliuspf** [comment]
I want to fix this first https://github.com/jasp-stats/INTERNAL-jasp/issues/1379

File: R/uniDimBayesianPreCalc.R [hunk 7]
```diff
@@ -62,7 +71,7 @@
     model[["n"]] <- nrow(dataset)
 
     # check if any items correlate negatively with the scale
-    model[["footnote"]] <- .checkLoadings(dataset, options[["variables"]])
+    model[["footnote"]] <- gettextf("%s %s", model[["footnote"]], .checkLoadings(dataset, options[["variables"]]))
```
Comment (vandenman) [inline R/uniDimBayesianPreCalc.R:74]: so, this implies that translators will be asked to translate
```r
%s %s
```
from one language to another. Do you think that makes sense? I'd use `sprintf` instead.

File: R/uniDimFrequentistPreCalc.R [hunk 8]
```diff
@@ -38,35 +38,44 @@
...
+    model[["n"]] <- n
     cc <- cov(dataset, use = model[["use.cases"]])
     model[["data_cov"]] <- cc
     model[["data_cor"]] <- cor(dataset, use = model[["use.cases"]])
     model[["itemsDropped"]] <- .unv(colnames(dataset))
 
 
     # check if any items correlate negatively with the scale
-    model[["footnote"]] <- .checkLoadings(dataset, options[["variables"]])
+    model[["footnote"]] <- gettextf("%s %s", model[["footnote"]], .checkLoadings(dataset, options[["variables"]]))
```
Comment (vandenman) [inline R/uniDimFrequentistPreCalc.R:78]: same thing here about translations

**Review by vandenman — CHANGES_REQUESTED** [review]

**Review by vandenman — APPROVED** [review]

**Pull request merged**